internal/application/service/applicationservice.go

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package service
+
+import (
+	"errors"
+
+	"github.com/asgardeo/thunder/internal/application/model"
+	"github.com/asgardeo/thunder/internal/system/config"
+	"github.com/asgardeo/thunder/internal/system/database"
+	"github.com/asgardeo/thunder/internal/system/log"
+
+	"go.uber.org/zap"
+)
+
+// ApplicationServiceInterface defines the interface for the application service.
+type ApplicationServiceInterface interface {
+	GetOAuthApplication(clientId string) (*model.OAuthApplication, error)
+}
+
```

Comment:
use 
```
var _ ApplicationServiceInterface = &ApplicationService{} 
```

This will validates whether the ApplicationService implements all the Interface functions in compile time

Reply:
Even without this, it validates whether ApplicationService implements all the Interface functions in compile time.

Ex: `go build` gives following error it missed to implement `GetOAuthApplication()`.
```
internal/application/service/applicationservice.go:46:9: cannot use &ApplicationService{} (value of type *ApplicationService) as ApplicationServiceInterface value in return statement: *ApplicationService does not implement ApplicationServiceInterface (missing method GetOAuthApplication)
```